app/helpers/ExerciseConfig/Compilation/VariablesResolver.php

```diff
@@ -120,8 +120,11 @@ public function resolveForInputNodes(MergeTree $mergeTree, VariablesTable $exerc
       $outputPort = current($inputBox->getOutputPorts());
       $variableName = $outputPort->getVariable();
       $child = current($node->getChildren());
-      $inputPortName = array_search($node, $child->getParents());
+      if ($child === false) {
+        throw new ExerciseConfigException("Input port not found for variable {$variableName}");
```

Comment:
Shouldn't this be handled by the validator?

Reply:
It definitely should, but was not... I ran into this error while testing PHP and better be sure that in there this cannot happen, I will create task for handling this in validator.

Reply:
OK, that sounds reasonable.